code/modules/hydroponics/plant_genes.dm

```diff
@@ -245,15 +251,50 @@
 	// For code, see grown.dm
 	name = "Liquid Contents"
 	examine_line = "<span class='info'>It has a lot of liquid contents inside.</span>"
+	trait_id = THROW_IMPACT_ID
 
-/datum/plant_gene/trait/squash/can_add(obj/item/seeds/S)
-	if(S.get_gene(/datum/plant_gene/trait/sticky))
-		return FALSE
+// Register a signal that our plant can be squashed on add.
+/datum/plant_gene/trait/squash/on_new(obj/item/food/grown/our_plant, newloc)
 	. = ..()
+	RegisterSignal(our_plant, COMSIG_PLANT_SQUASH, .proc/squash_plant)
+
+// Squash the plant on slip.
+/datum/plant_gene/trait/squash/on_slip(obj/item/food/grown/our_plant, mob/living/carbon/target)
+	SEND_SIGNAL(our_plant, COMSIG_PLANT_SQUASH, target)
+
+// Squash the plant on thrown impact.
+/datum/plant_gene/trait/squash/on_throw_impact(obj/item/food/grown/our_plant, atom/target)
+	SEND_SIGNAL(our_plant, COMSIG_PLANT_SQUASH, target)
+
+/datum/plant_gene/trait/squash/proc/squash_plant(obj/item/food/grown/our_plant, atom/target)
+	SIGNAL_HANDLER
+
+	var/turf/our_turf = get_turf(target)
+	our_plant.forceMove(our_turf)
+	if(istype(our_plant))
+		if(ispath(our_plant.splat_type, /obj/effect/decal/cleanable/food/plant_smudge))
+			var/obj/plant_smudge = new our_plant.splat_type(our_turf)
+			plant_smudge.name = "[our_plant.name] smudge"
+			if(our_plant.filling_color)
+				plant_smudge.color = our_plant.filling_color
+		else if(our_plant.splat_type)
+			new our_plant.splat_type(our_turf)
+	else
+		var/obj/effect/decal/cleanable/food/plant_smudge/misc_smudge = new(our_turf)
+		misc_smudge.name = "[our_plant.name] smudge"
+		misc_smudge.color = "#82b900"
+
+	our_plant.visible_message("<span class='warning'>[our_plant] is squashed.</span>","<span class='hear'>You hear a smack.</span>")
+	if(IS_PLANT(our_plant))
+		if(our_plant.seed)
```

Comment:
I still don't really like this--this still relies on `seed` being implemented on *both* type paths, when only one is enforced at compile time. 

You have two options. One is to remove `IS_PLANT`, and instead check both types specifically, and get their `.seed` from that.

The other is to create a more generic solution, such as storing seeds in the contents themselves, and looping through them at this part of the process.

Apparently the latter contributes to map tick, and the first one is much simpler.

Reply:
Hydroponics and plant code does this weird `.seed` thing for both types of plants all over the place, i've been thinking about ways to refactor plants so it stops requiring it but I haven't figured anything simple out yet. I'll think on this more for now.